gridscale/service-query/blocker.go

```diff
@@ -0,0 +1,140 @@
+package service_query
+
+import (
+	"fmt"
+	"github.com/gridscale/gsclient-go"
+	"github.com/hashicorp/terraform/helper/resource"
+	"time"
+)
+
+const (
+	delayFetchingStatus  = 500 * time.Millisecond
+	timeoutCheckDeletion = 1 * time.Minute
+)
+
+const (
+	provivisoningStatus = "in-provisioning"
+	activeStatus        = "active"
+)
+
+type gsService string
+
+const (
+	LoadbalancerService gsService = "loadbalancer"
+	IPService           gsService = "IP"
+	NetworkService      gsService = "network"
+	ServerService       gsService = "server"
+	SSHKeyService       gsService = "sshkey"
+	StorageService      gsService = "storage"
+	ISOImageService     gsService = "isoimage"
+)
+
+//BlockProvisoning blocks until the object's state is not in provisioning anymore
+func BlockProvisoning(client *gsclient.Client, service gsService, id string, timeout time.Duration) error {
```

Comment:
I would prefer `RetryUntilResourceStatusIsActive` to reflect what the function does.